appservice/src/createAppService/SiteNameStep.ts

```diff
@@ -122,12 +127,104 @@ export class SiteNameStep extends AzureNameStep<SiteNameStepWizardContext> {
+            validationMessage ??= await this.asyncValidateUniqueARMId(context, sdkClient, name, context.resourceGroup?.name ?? context.newResourceGroupName);
+        }
+
+        return validationMessage;
+    }
+
+    private async asyncValidateGlobalCNA(client: WebSiteManagementClient, name: string): Promise<string | undefined> {
         const nameAvailability: ResourceNameAvailability = await client.checkNameAvailability(name, 'Site');
         if (!nameAvailability.nameAvailable) {
             return nameAvailability.message;
         } else {
             return undefined;
         }
     }
+
+    private async asyncValidateRegionalCNA(context: SiteNameStepWizardContext, domainNameScope: DomainNameLabelScope, siteName: string, resourceGroupName?: string): Promise<string | undefined> {
+        if (!LocationListStep.hasLocation(context)) {
+            throw new Error(vscode.l10n.t('Internal Error: A location is required when validating a site name with regional CNA.'));
+        }
+        if (domainNameScope === DomainNameLabelScope.ResourceGroup && !resourceGroupName) {
+            throw new Error(vscode.l10n.t('Internal Error: A resource group name is required for validating this level of domain name scope.'));
+        }
+
+        const apiVersion: string = '2024-04-01';
+        const location: AzExtLocation = await LocationListStep.getLocation(context);
+        const authToken: string = nonNullValueAndProp((await context.credentials.getToken() as { token?: string }), 'token');
+
+        // Todo: Can replace with call using SDK once the update is available
+        const options: AzExtRequestPrepareOptions = {
+            url: `https://management.azure.com/subscriptions/${context.subscriptionId}/providers/Microsoft.Web/locations/${location.name}/checknameavailability?api-version=${apiVersion}`,
+            method: 'POST',
+            headers: createHttpHeaders({
+                'Content-Type': 'application/json',
+                'Authorization': `Bearer ${authToken}`,
+            }),
+            body: JSON.stringify({
+                name: siteName,
+                type: 'Site',
+                autoGeneratedDomainNameLabelScope: domainNameScope,
+                resourceGroupName: domainNameScope === DomainNameLabelScope.ResourceGroup ? resourceGroupName : undefined,
+            }),
+        };
+
+        const client = await createGenericClient(context, undefined);
+        const pipelineResponse = await client.sendRequest(createPipelineRequest(options)) as AzExtPipelineResponse;
+        const checkNameResponse = pipelineResponse.parsedBody as {
+            hostName?: string;
+            message?: string;
+            nameAvailable?: boolean;
+            reason?: string;
+        };
+
+        if (!checkNameResponse.nameAvailable) {
+            // If site name input is >=47 chars, ignore result of regional CNA because it inherently has a shorter character limit than Global CNA
```

Comment:
Should you change `validateSiteName` to check the max length based on regional and global then?